src/vellum/workflows/monitoring/context.py

```diff
@@ -0,0 +1,206 @@
+"""Monitoring execution context for workflow tracing."""
+
+from contextlib import contextmanager
+from dataclasses import field
+import threading
+from uuid import UUID
+from typing import Dict, Iterator, Optional
+
+from vellum.client.core import UniversalBaseModel
+from vellum.workflows.context import get_execution_context
+from vellum.workflows.events.types import ParentContext
+
+
+class MonitoringExecutionContext(UniversalBaseModel):
+    """Monitoring-specific execution context that runs in parallel with ExecutionContext."""
+
+    trace_id: UUID = field(default_factory=lambda: UUID("00000000-0000-0000-0000-000000000000"))
+    parent_context: Optional[ParentContext] = None
+    monitoring_enabled: bool = True
```

Comment:
do we need to introduce _another_ context object in order to do this forked path and maintain both? especially since `monitoring_enabled` should never be false here correct?

Reply:
Gonna do the rest of the review on the other PR, I think we could save needing to introduce this one